src/app/src/main/java/com/example/bookmark/ScanIsbnActivity.java

```diff
@@ -1,18 +1,226 @@
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_scan_isbn);
+        getSupportActionBar().setTitle("Scan ISBN");
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+
+        mPreviewView = findViewById(R.id.viewFinder);
+
+        BarcodeScannerOptions options =
+            new BarcodeScannerOptions.Builder()
+                .setBarcodeFormats(
+                    Barcode.FORMAT_EAN_13, // Format of ISBN barcode's
+                    Barcode.FORMAT_EAN_8)
+                .build();
+        scanner = BarcodeScanning.getClient();
+
+        if(allPermissionsGranted()){
+            startCamera(); //start camera if permission has been granted by user
+        } else {
+            Log.d("PHOTO", "Should request permissions");
+            ActivityCompat.requestPermissions(this, REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS);
+        }
+    }
+
+    private void startCamera() {
+
+        final ListenableFuture<ProcessCameraProvider> cameraProviderFuture = ProcessCameraProvider.getInstance(this);
+
+        cameraProviderFuture.addListener(new Runnable() {
+            @Override
+            public void run() {
+                try {
+
+                    ProcessCameraProvider cameraProvider = cameraProviderFuture.get();
+                    bindPreview(cameraProvider);
+
+                } catch (ExecutionException | InterruptedException e) {
+                    // No errors need to be handled for this Future.
+                    // This should never be reached.
+                }
+            }
+        }, ContextCompat.getMainExecutor(this));
+    }
+
+    void bindPreview(@NonNull ProcessCameraProvider cameraProvider) {
+
+        cameraProvider.unbindAll();
+
+        Preview preview = new Preview.Builder()
+            .setTargetName("viewFinder")
+            .build();
+
+        CameraSelector cameraSelector = new CameraSelector.Builder()
+            .requireLensFacing(CameraSelector.LENS_FACING_BACK)
+            .build();
+
+        ImageAnalysis imageAnalysis =
+            new ImageAnalysis.Builder()
+                .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+                .build();
+
+        imageAnalysis.setAnalyzer(executor, new ImageAnalysis.Analyzer() {
+            @Override
+            public void analyze(@NonNull ImageProxy imageProxy) {
+                analyzeBarcode(imageProxy);
+            }
+        });
+
+        try{
+            Camera camera = cameraProvider.bindToLifecycle(this, cameraSelector, preview, imageAnalysis);
+            preview.setSurfaceProvider(mPreviewView.createSurfaceProvider());
+        } catch(Exception e){
+            Log.e("error", String.valueOf(e));
+        }
+    }
+
+    private void analyzeBarcode(ImageProxy imageProxy) {
+        @SuppressLint("UnsafeExperimentalUsageError") Image mediaImage = imageProxy.getImage();
```

Comment:
Kinda curious about why we're turning off "UnsafeExperimentalUsageError" here.

Reply:
The library is in beta